scripts/o3de/o3de/compatibility.py

```diff
@@ -70,7 +70,7 @@ def get_incompatible_gem_dependencies(gem_json_data:dict, all_gems_json_data:dic
     if not gem_dependencies:
         return set()
 
-    return get_incompatible_gem_version_specifiers(gem_dependencies, all_gems_json_data)
+    return get_incompatible_gem_version_specifiers(gem_dependencies, all_gems_json_data, set())
```

Comment:
Is the empty ```set()``` argument needed here if ```checked_specifiers``` already defaults to ```None```? You could also do named arguments here as well
```suggestion
    return get_incompatible_gem_version_specifiers(gem_version_specifier_list=gem_dependencies, 
                                        all_gems_json_data=all_gems_json_data)
```